src/zenml/stack_stores/local_stack_store.py

```diff
@@ -187,6 +187,94 @@ def register_stack_component(
+    ) -> None:
+        """Update a stack component."""
+        components = self.__store.stack_components[component.type]
+        if component.name not in components:
+            raise StackComponentExistsError(
+                f"Unable to update stack component (type: {component.type}) "
+                f"with name '{component.name}': No existing stack component "
+                f"found with this name."
+            )
+
+        # write the component configuration file
+        component_config_path = self._get_stack_component_config_path(
+            component_type=component.type, name=component.name
+        )
+        utils.create_dir_recursive_if_not_exists(
+            os.path.dirname(component_config_path)
+        )
+        utils.write_file_contents_as_string(
+            component_config_path,
+            base64.b64decode(component.config).decode(),
+        )
+
+        # add the component to the stack store dict and write it to disk
+        components[component.name] = component.flavor
+        self._write_store()
+        logger.info(
+            "Updated stack component with type '%s' and name '%s'.",
+            component.type,
+            component.name,
+        )
+
+    def rename_stack_component(
+        self,
+        old_name: str,
+        component: StackComponentWrapper,
+    ) -> None:
+        """Rename a stack component."""
+        components = self.__store.stack_components[component.type]
+        if old_name not in components:
+            raise StackComponentExistsError(
+                f"Unable to rename stack component (type: {component.type}) "
+                f"with name '{component.name}': No existing stack component "
+                f"found with this name."
+            )
+
+        # write the component configuration file
+        renamed_component_config_path = self._get_stack_component_config_path(
+            component_type=component.type, name=component.name
+        )
+        utils.create_dir_recursive_if_not_exists(
+            os.path.dirname(renamed_component_config_path)
+        )
+        utils.write_file_contents_as_string(
+            renamed_component_config_path,
+            base64.b64decode(component.config).decode(),
+        )
+
+        # delete the old stack component
+        self._delete_stack_component(component.type, old_name)
+
+        # add the component to the stack store dict and write it to disk
+        components[component.name] = component.flavor
+        self._write_store()
+        logger.info(
+            "Renamed stack component with type '%s' and new name '%s'.",
+            component.type,
+            component.name,
+        )
+
+    def update_stacks_after_rename(
+        self,
+        old_name: str,
+        new_name: str,
+        renamed_component_type: StackComponentType,
+    ) -> None:
+        """Update stack components on stacks following a component rename."""
+        for _, conf in self.stack_configurations.items():
```

Comment:
Do you think it makes sense to include this logic in the stack component renaming? I couldn't think of any situation in which we would want to rename a stack componetn but leave the old names inside the stacks, so I feel it's going to be safer to just handle it in one self-sufficient call instead.

Reply:
Yeah this has been completely overhauled now.